File: 프로그래머스/lv2/연속 부분 수열 합의 개수.py
def solution(elements):
    dp=[[0]*(len(elements)) for _ in range(len(elements))]
    for i in range(0,len(elements)):
        for j in range(i,len(elements)+i):
            dp[i][j%len(elements)]=dp[i-1][j%len(elements)]+elements[(j+i)%len(elements)]
    total=[]
    for i in dp:
        total+=set(i)
    answer=len(set(total))
    return answer

print(solution([7,9,1,1,4]))

File: 프로그래머스/lv2/연속부분수열합의개수.java
import java.util.HashSet;

public class 연속부분수열합의개수 {
    public int solution(int[] elements) {
        HashSet<Integer> result = new HashSet<>();
        for(int i=0;i<elements.length;i++){
            int ssum=elements[i];
            result.add(ssum);
            for(int j=i+1;j<i+elements.length;j++){
                ssum+=elements[j%elements.length];
                result.add(ssum);
            }
        }
        return result.size();
    }

    public static void main(String[] args) {
        연속부분수열합의개수 sum_N = new 연속부분수열합의개수();
        System.out.println(sum_N.solution(new int[]{7, 9, 1, 1, 4}));
    }
}
